src/metabase/models/params.clj

```diff
@@ -68,6 +69,28 @@
     :else
     field-id-or-form))
 
+(def ^:dynamic *ignore-current-user-perms-and-return-all-field-values*
+  "Whether to ignore permissions for the current User and return *all* FieldValues for the Fields being parameterized by
+  Cards and Dashboards. This determines how `:param_values` gets hydrated for Card and Dashboard. Normally, this is
+  `false`, but the public and embed versions of the API endpoints can bind this to `true` to bypass normal perms
+  checks (since there is no current User) and get *all* values."
+  false)
+
+(defn- field-ids->param-field-values-ignoring-current-user
+  [param-field-ids]
+  (t2/select-fn->fn :field_id identity ['FieldValues :values :human_readable_values :field_id]
+                    :type :full
+                    :field_id [:in param-field-ids]))
+
+(defn- field-ids->param-field-values
+  "Given a collection of `param-field-ids` return a map of FieldValues for the Fields they reference. This map is
+  returned by various endpoints as `:param_values`."
```

Comment:
Consider documenting that `nil` is returned if `param-field-ids` is empty.

Reply:
I documented it in https://github.com/metabase/metabase/pull/29359/commits/46c41746948627cd8e6aab89464f227162332464